### PR TITLE
Enable and fix `node.unconfirmed_send` unit test

### PR DESCRIPTION
**Pull request by theohax**
I couldn't find an issue for this unit test to link it to the PR, perhaps it was disabled way back.

### CODE REVIEW

File: nano/core_test/node.cpp [hunk 1]
```diff
@@ -1682,25 +1682,43 @@ TEST (node, bootstrap_confirm_frontiers)
 }
 
 // Test that if we create a block that isn't confirmed, we sync.
-TEST (node, DISABLED_unconfirmed_send)
+TEST (node, unconfirmed_send)
```
Comment (dsiganos) [inline nano/core_test/node.cpp:1685]: Could you write an overview paragraph here explaining what the test does overall?

Comment (dsiganos) [reply nano/core_test/node.cpp:1685]: On seconds thoughts,  it is fine, the inline comments are good enough to understand what is happening.

File: nano/core_test/node.cpp [hunk 2]
```diff
@@ -1682,25 +1682,43 @@ TEST (node, bootstrap_confirm_frontiers)
...
+
+	nano::keypair key1{};
+	auto & node1 = *system.nodes[1];
+	auto wallet1 = system.wallet (1);
+	wallet1->insert_adhoc (key1.prv);
+
+	// firstly, send two units from node0 to node1 and expect they're both received (as balance) on the other end
+	//
+	auto send1 = wallet0->send_action (nano::dev::genesis->account (), key1.pub, 2 * nano::Mxrb_ratio);
+	ASSERT_TIMELY (10s, node1.balance (key1.pub) == 2 * nano::Mxrb_ratio && !node1.bootstrap_initiator.in_progress ());
```
Comment (dsiganos) [inline nano/core_test/node.cpp]: Why does it matter if bootstrap initiator is in progress?

File: nano/core_test/node.cpp [hunk 3]
```diff
@@ -1682,25 +1682,43 @@ TEST (node, bootstrap_confirm_frontiers)
...
+				 .previous (latest1)
+				 .representative (nano::dev::genesis_key.pub)
+				 .balance (nano::Mxrb_ratio)
+				 .link (nano::dev::genesis->account ())
+				 .sign (key1.prv, key1.pub)
+				 .work (*system.work.generate (latest1))
+				 .build_shared ();
+	ASSERT_EQ (nano::process_result::progress, node1.ledger.process (node1.store.tx_begin_write (), *send2).code);
+	node1.process_confirmed (nano::election_status{ send2 });
+	ASSERT_TIMELY (5s, node1.block_confirmed (send2->hash ()));
```
Comment (dsiganos) [inline nano/core_test/node.cpp]: Does node0 get synced at this point?
Is there an election taking place at this point?
I think an overall comment describing those details would be good afterall.

**Comment by theohax** [comment]
@reviewers -- addressed CR, please re-review, thanks

File: nano/core_test/node.cpp [hunk 4]
```diff
@@ -1682,26 +1682,40 @@ TEST (node, bootstrap_confirm_frontiers)
 }
 
 // Test that if we create a block that isn't confirmed, we sync.
-TEST (node, DISABLED_unconfirmed_send)
+TEST (node, unconfirmed_send)
```
Comment (dsiganos) [inline nano/core_test/node.cpp:1685]: What is the intention of this test? What is it trying to test?
It is called unconfirmed send but I do not see anything unconfirmed in the test...
Which block is unconfirmed?

File: nano/core_test/node.cpp [hunk 5]
```diff
@@ -1682,26 +1682,40 @@ TEST (node, bootstrap_confirm_frontiers)
...
+	auto send1 = wallet1->send_action (nano::dev::genesis->account (), key2.pub, 2 * nano::Mxrb_ratio);
+	ASSERT_TIMELY (5s, node1.block_confirmed (send1->hash ()));
+	ASSERT_TIMELY (5s, node2.block_confirmed (send1->hash ()));
+	ASSERT_TIMELY (10s, node2.balance (key2.pub) == 2 * nano::Mxrb_ratio);
+
+	// then send one of the received units back from node2 to node1;
+	//
+	auto const latest = node2.latest (key2.pub);
+	auto send2 = nano::state_block_builder{}.make_block ().account (key2.pub).previous (latest).representative (nano::dev::genesis_key.pub).balance (nano::Mxrb_ratio).link (nano::dev::genesis->account ()).sign (key2.prv, key2.pub).work (*system.work.generate (latest)).build_shared ();
+	node2.process_active (send2);
```
Comment (dsiganos) [inline nano/core_test/node.cpp]: I think that this should not be a process_active. I think this should just insert a block into the ledger bypassing block processing. That is probably what is meant by "unconfirmed send". The original unit test calls `node1.ledger.process`

Comment (theohax) [reply nano/core_test/node.cpp]: Not sure about that to be honest but I addressed it by using `node.process` now.

**Comment by theohax** [comment]
@dsiganos @clemahieu addressed CR guys, pinging you once more to have a look at this, thanks

**Review by dsiganos — APPROVED** [review]

**Comment by theohax** [comment]
LGTM, good to merge

**Pull request merged**